### PR TITLE
[MODULAR-ISH] Entombed Quirk from Nova Sector

### PR DESCRIPTION
**Pull request by Superhats**
## About The Pull Request
Ports the entombed quirk from Nova Sector (https://github.com/NovaSector/NovaSector/pull/656)
Almost fully modular except for a couple lines in modular_skyrat/.../equipping.dm, I'll gladly make it fully modular if someone can figure out how.

Tl;dr from the original PR: Quirk that gives you a weaker-than-civilian (Tier 3 armor, 10 complexity), unremovable modsuit that slowly kills you if depowered. Players can customize the suit's name and description, as well as a prefix on each limb.
Optionally impossible to retract, forcing Medical to fix you using tools like the thread ripper and piercing syringes. 

Notes: As-is, this allows players to pick the infiltrator and interdyne modsuits as skins (Completely cosmetic), I will also gladly snip those out if deemed problematic. The original PR also includes the cosmohonk suit as a skin, which I've already removed.

I'll also preemtively address one potential concern: Due to the suit being unremovable, attempts to strip it will empty the backpack's contents, still allowing people to access what's inside.

## Why It's Good For The Game
Echoing the original PR, this opens the door to a lot of interesting characters, whether you're a sentient modsuit puppetting a corpse in the vein of Fallout's Y-17 trauma harness, or just some unfortunate guy in an iron lung.

## Proof Of Testing
image
image
image
image

## Changelog

:cl: VrazzleDazzle, yooriss
add: Added the Entombed quirk from Nova Sector
/:cl:

### CODE REVIEW

File: modular_zubbers/master_files/code/modules/entombed_quirk/code/entombed_mod.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		/obj/item/mod/module/joint_torsion/entombed,
+		/obj/item/mod/module/storage,
+	)
+	allowed_suit_storage = list(
+		/obj/item/tank/internals,
+		/obj/item/flashlight,
+	)
+
+/datum/armor/mod_entombed
+	melee = ARMOR_LEVEL_WEAK
```
Comment (YakumoChen) [inline modular_zubbers/master_files/code/modules/entombed_quirk/code/entombed_mod.dm:25]: Use values, not these defines. These defines only exist to make it harder to read the code

**Comment by YakumoChen** [comment]
30 armour across the board is pretty strong, better than actual security options, and you should probably reduce most of them to 5 or 10 (or remove them entirely)

**Comment by Superhats** [comment]
I've now nerfed it to the same values as the civilian MODsuit. Which, paired with it having 5 less complexity, still makes it a distinct downgrade.

**Comment by IgiariValkyr** [comment]
> I've now nerfed it to the same values as the civilian MODsuit. Which, paired with it having 5 less complexity, still makes it a distinct downgrade.
> 
> That said if people think that's _still_ too much, I'm willing to just give it 0 armor across the board (except for Bio)

Mmm... Honestly, the armor values look fine now to me. It's supposed to be weaker than a standard modsuit, which it is now, but still able to have protection for EVA and light debris like a metal suit would. Or in this case, metal tomb.

**Comment by nevimer** [comment]
> I've now nerfed it to the same values as the civilian MODsuit. Which, paired with it having 5 less complexity, still makes it a distinct downgrade.
> 
> That said if people think that's _still_ too much, I'm willing to just give it 0 armor across the board (except for Bio)

I am fine with 30 across the board given that being trapped in it as a pretty big downside. Keep the armour.

**Comment by Superhats** [comment]
Armor's back at 30, except bio at 100 to keep it consistent since it's supposed to be a fully sealed suit.

**Comment by projectkepler-ru** [comment]
> 30 armour across the board is pretty strong, better than actual security options, and you should probably reduce them to 5 (or remove them entirely)

didn't you nerf the sec options?

**Comment by projectkepler-ru** [comment]
Why do you have AI highlighted that greatly upsets me

**Comment by Superhats** [comment]
I PLAYED AI ONCE AND I KEEP TRYING TO REMOVE IT BUT I CAN'T

**Comment by YakumoChen** [comment]
> > 30 armour across the board is pretty strong, better than actual security options, and you should probably reduce them to 5 (or remove them entirely)
> 
> didn't you nerf the sec options?

Yes, I did.

I don't agree the suit should have high armour. It is a downside to wear, it shouldn't come with upsides.

**Comment by Superhats** [comment]
The armor is arguably the only upside of this suit, when you consider a standard MOD isn't exactly hard to get early on, plating/cores are very cheap to just buy and the shell itself only costs iron and maybe a tiny bit of plasma.

It's got less complexity than every other MOD in the game, permanently takes up your backpack slot which forces you to use even more of your already limited complexity on a storage module, and if you tick the box that makes it unretractable, you're impossible to fix unless medbay's very competent.

**Comment by Majkl-J** [comment]
You're very easy to fix as long as anyone has a threadripper. Which the CMO starts with.

**Comment by Superhats** [comment]
> You're very easy to fix as long as anyone has a threadripper. Which the CMO starts with.

The CMO doesn't actually start with a threadripper. The rescue suit is advanced quick carry and injector.

**Comment by Superhats** [comment]
Now granted, it's a fairly popular purchase nowadays, but it's still _a factor._

But also, I'm not hugely attached to the armor myself and only put it back because @nevimer told me to. I can revert that if it's what's holding the PR up.

**Comment by Majkl-J** [comment]
I'm not against the armor values myself, we can drop them down later, but damn guess I automated getting the thread ripper so much I started believing the CMO gets it roundstart

**Comment by Superhats** [comment]
Either way could this get a TM maybe? I feel like there's not much to do besides speculate now without seeing how people actually use the quirk.

**Comment by NanoCats** [comment]
We have clowns, and if I choose to grief myself as a clown with this quirk, why not? Put it back :c 

Also the slowdown feels like a bit much trying it out really quick.

**Comment by Superhats** [comment]
Changelog: You can now replace the storage module. Also the cosmohonk's back and I put the "Makes you really hard to revive" button at the top

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]
Ok, I like the quirk. But there's just one glaring issue.

Non-removable storage is something we learned very quickly on #625 and is not going to be accepted. There needs to be SOME way to dump the storage. Either it be something like snail code, or something else entirely. Unremovable back slot storage is not something we're going to repeat.

**Comment by Superhats** [comment]
> Ok, I like the quirk. But there's just one glaring issue.
> 
> Non-removable storage is something we learned very quickly on #625 and is not going to be accepted. There needs to be SOME way to dump the storage. Either it be something like snail code, or something else entirely. Unremovable back slot storage is not something we're going to repeat.

That's already a thing and I've already addressed it in the original description. Just like snails, trying to remove the suit will dump it out.

**Comment by StrangeWeirdKitten** [comment]
> > Ok, I like the quirk. But there's just one glaring issue.
> > Non-removable storage is something we learned very quickly on #625 and is not going to be accepted. There needs to be SOME way to dump the storage. Either it be something like snail code, or something else entirely. Unremovable back slot storage is not something we're going to repeat.
> 
> That's already a thing and I've already addressed it in the original description. Just like snails, trying to remove the suit will dump it out.

Ok that's based, thanks for clearifying. nvm then.

**Review by StrangeWeirdKitten — APPROVED** [review]

**Comment by Superhats** [comment]
I've removed the infiltrator suit from the sprites since it looks identical to a suit actual antags can wear on station and has apparently caused some troubles with security.

**Comment by nevimer** [comment]
maybe it can return via emag_act?

**Comment by Superhats** [comment]
I have no idea how to do that? Given it's like, just a skin slapped onto a different suit, it doesn't actually have infiltrator stats or anything.

**Comment by GoreDem** [comment]
If you get fullstripped (admin, bloodsucker abilities, gulag teleporter) you cannot re-equip the mod (because it gets glued to your hand) and you just die, with no possible IC fix.

**Comment by YakumoChen** [comment]
> If you get fullstripped (admin, bloodsucker abilities, gulag teleporter) you cannot re-equip the mod (because it gets glued to your hand) and you just die, with no possible IC fix.

aside from admin force-strip, these are bugs and unrelated to problems with the MODsuit

NO_DROP backpacks can be equipped from hand by clicking directly on the slot to be equipped on, pressing the hotkey does not work

**Comment by NanoCats** [comment]
Question: can we have it be so that if I click my expand/retract suit button it does the helmet. like if I can only do the helmet and only manually, do that when I press the normal button.  Just a little thing.

**Comment by Superhats** [comment]
I'm probably going to end up closing this given it runtimes to oblivion and the reception is generally negative for anyone _not_ playing the entombed gamer.

**Pull request merged**